fenix/app/src/main/java/org/mozilla/fenix/search/awesomebar/AwesomeBarView.kt

```diff
@@ -278,45 +283,77 @@ class AwesomeBarView(
         }
 
         if (state.showAllHistorySuggestions) {
-            if (activity.settings().historyMetadataUIFeature) {
-                providersToAdd.add(defaultCombinedHistoryProvider)
-            } else {
-                providersToAdd.add(defaultHistoryStorageProvider)
-            }
+            providersToAdd.add(
+                getHistoryProvider(
+                    filters = listOfNotNull(getFilterToExcludeSponsoredSuggestions(state)),
+                ),
+            )
         }
 
         if (state.showHistorySuggestionsForCurrentEngine) {
-            getHistoryProvidersForSearchEngine(state.searchEngineSource)?.let {
-                providersToAdd.add(it)
-            }
+            providersToAdd.add(
+                getHistoryProvider(
+                    filters = listOfNotNull(
+                        getFilterToExcludeSponsoredSuggestions(state),
```

Comment:
This will always be `null` here, because we never show sponsored suggestions when a shortcut engine is selected—IOW, there's never a case where `state.show*SuggestionsForCurrentEngine == true && state.showSponsoredSuggestions == true` today—but I kept it for symmetry with the `showAll*Suggestions` cases.

Reply:
Sounds good, that makes sense!